From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Luigi Pinca <luigipinca@gmail.com>
Date: Wed, 13 Sep 2023 08:04:39 +0200
Subject: test: deflake test-tls-socket-close

Move the check for the destroyed state of the remote socket to the inner
`setImmediate()`.

Refs: https://github.com/nodejs/node/pull/49327#issuecomment-1712525257
PR-URL: https://github.com/nodejs/node/pull/49575
Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>
Reviewed-By: Moshe Atlow <moshe@atlow.co.il>

diff --git a/test/parallel/test-tls-socket-close.js b/test/parallel/test-tls-socket-close.js
index 667b291309a4c5636a2c658fa8204b32c2e4df46..70af760d53bb4ddab62c99180d505e943ec269f6 100644
--- a/test/parallel/test-tls-socket-close.js
+++ b/test/parallel/test-tls-socket-close.js
@@ -44,9 +44,9 @@ function connectClient(server) {
 
       setImmediate(() => {
         assert.strictEqual(netSocket.destroyed, true);
-        assert.strictEqual(clientTlsSocket.destroyed, true);
 
         setImmediate(() => {
+          assert.strictEqual(clientTlsSocket.destroyed, true);
           assert.strictEqual(serverTlsSocket.destroyed, true);
 
           tlsServer.close();
